pkgs/development/libraries/mesa/default.nix

```diff
@@ -90,124 +89,126 @@ let self = stdenv.mkDerivation {
-    "--enable-llvm-shared-libs"
-    (enableFeature stdenv.isLinux "omx-bellagio")
-    (enableFeature stdenv.isLinux "va")
-    "--disable-opencl"
-  ];
-
-  nativeBuildInputs = [
-    autoreconfHook intltool pkgconfig file
-    python2 python2Packages.Mako
-  ];
-
-  propagatedBuildInputs = with xorg; [
-    libXdamage libXxf86vm
-  ] ++ optional stdenv.isLinux libdrm
-    ++ optionals stdenv.isDarwin [ OpenGL Xplugin ];
+    # "localstatedir=\${TMPDIR}" # previous install flag differs?
+
+    # Using "drivers" results in `cycle detected in the references of $drivers from $out`
+    "-Ddri-drivers-path=${placeholder "out"}/lib/dri" # "--with-dri-driverdir"
+    "-Dvdpau-libs-path=${placeholder "out"}/lib/vdpau"
+    "-Dxvmc-libs-path=${placeholder "out"}/lib"
+    "-Domx-libs-path=${placeholder "out"}/lib/bellagio"
+    "-Dva-libs-path=${placeholder "out"}/lib/dri"
+    "-Dd3d-drivers-path=${placeholder "out"}/lib/d3d"
+
+    "-Ddri-search-path=${libglvnd.driverLink}/lib/dri" # "--with-dri-searchpath"
+    "-Dplatforms=${concatStringsSep "," eglPlatforms}" # "--with-platforms"
+    "-Dgallium-drivers=${concatStringsSep "," galliumDrivers}" # "--with-gallium-drivers"
+    "-Ddri-drivers=${concatStringsSep "," driDrivers}" # "--with-dri-drivers"
+    "-Dvulkan-drivers=${concatStringsSep "," vulkanDrivers}" # "--with-vulkan-drivers"
+    # "--enable-texture-float" # outdated...?
+    "-Dglvnd=true" # "--enable-libglvnd"
+    "-Dglx-direct=true" # "--enable-driglx-direct"
+    "-Dgles1=true" # "--enable-gles1"
+    "-Dgles2=true" # "--enable-gles2"
+    "-Dglx=dri" # "--enable-glx" + "--enable-dri" https://gitlab.freedesktop.org/mesa/mesa/blob/master/configure.ac#L1464
+    "-Dllvm=true" # "--enable-llvm"
+    "-Dgallium-xvmc=true" # "--enable-xvmc"
+    "-Dgallium-vdpau=true" # "--enable-vdpau"
+    "-Dshared-glapi=true" # "--enable-shared-glapi"
+    "-Dshared-llvm=true" # "--enable-llvm-shared-libs"
+    "-Dgallium-opencl=disabled" # "--disable-opencl"
+  ] ++ optional (elem "swrast" galliumDrivers) "-Dosmesa=gallium" # used by wine
+    # (enableFeature (elem "swrast" galliumDrivers) "gallium-osmesa")
+    # ++ optional (!stdenv.hostPlatform.isMusl) "" # https://bugs.freedesktop.org/show_bug.cgi?id=35268 # autotools thing...?
+    # (enableFeature (!stdenv.hostPlatform.isMusl) "glx-tls")
+    ++ optionals (stdenv.isLinux) [
+      "-Ddri3=true"
+      # (enableFeature stdenv.isLinux "dri3")
+      "-Dgallium-nine=true"
+      # (enableFeature stdenv.isLinux "nine") # Direct3D in Wine
+      "-Degl=true"
+      # (enableFeature stdenv.isLinux "egl")
+      "-Dgallium-xa=true" # used in vmware driver
+      # (enableFeature stdenv.isLinux "xa")
+      "-Dgbm=true"
+      # (enableFeature stdenv.isLinux "gbm")
+      "-Dgallium-omx=bellagio"
+      # (enableFeature stdenv.isLinux "omx-bellagio")
+      "-Dgallium-va=true"
+      # (enableFeature stdenv.isLinux "va")
+    ];
 
   buildInputs = with xorg; [
+    flex bison
```

Comment:
These should probably go to `nativeBuildInputs` too if they are executed during build.

Reply:
Like https://gitlab.freedesktop.org/mesa/mesa/blob/master/src/mesa/program/meson.build#L32?